docs/android/v2/how--to-guides/interact-with-room/room/session-metadata.mdx

```diff
@@ -0,0 +1,106 @@
+ nav: 7.24
+---
+
+Session metadata is a shared state that is accessible to all participants in a room. It can be used for features such as pinning text or highlighting a particular participant. The session metadata is retained throughout the duration of a session and is cleared once all participants have left the room.
+
+## Differences between Session Metadata and Peer Metadata
+
+While [peer metadata](https://www.100ms.live/docs/android/v2/how--to-guides/interact-with-room/peer/peer-metadata-update) is associated with individual peers and each peer can have their own metadata, session metadata remains the same for every peer in the room.
+
+## How to store shared data with 100ms SDK
+
+To update the value of the session metadata for the current room session, call `setSessionMetadata` method on SDK instance. In the following example, the "test" string will be saved in the session metadata and accessible to all participants in the room as a shared state.
+
+Make sure you call session metadata API after you joined the room. You should listen to [onJoin()](https://www.100ms.live/docs/android/v2/how--to-guides/set-up-video-conferencing/join#call-join) callback to proceed.
+
+<Tabs id="setsessionmetadata" items={['Kotlin', 'Java']} />
+
+<Tab id='setsessionmetadata-0'>
+
+```kotlin
+hmsSDK.setSessionMetaData("some data", object : HMSActionResultListener {
+      override fun onError(error: HMSException) {
+      }
+
+      override fun onSuccess() {          
+      }
+})
+```
+
+</Tab>
+
+<Tab id='setsessionmetadata-1'>
+
+```java
+hmsSdk.setSessionMetaData("some data", new HMSActionResultListener() {
+    @Override
+    public void onSuccess() {
+    }
+
+    @Override
+    public void onError(@NonNull HMSException e) {
+    }
+});
+
+```
+
+</Tab>
+
+## How to read shared data with 100ms SDK
+
+To retrieve the session metadata, use the `getSessionMetadata` method:
+
+Make sure you call session metadata API after you joined the room. You should listen to [onJoin()](https://www.100ms.live/docs/ios/v2/how--to-guides/set-up-video-conferencing/join#call-join) callback to proceed.
+
+<Tabs id="getsessionmetadata" items={['Kotlin', 'Java']} />
+
+<Tab id='getsessionmetadata-0'>
+
+```kotlin
+hmsSDK.getSessionMetaData(object :HMSSessionMetadataListener {
+    override fun onSuccess(sessionMetadata: String?) {
+        // Use the metadata
+        Log.d("Data","The data is $sessionMetadata")
+    }
+
+    override fun onError(error: HMSException) {
+    }
+})
+```
+
+</Tab>
+
+<Tab id='getsessionmetadata-1'>
+
+```java
+hmsSdk.getSessionMetaData(new HMSSessionMetadataListener() {
+    @Override
+    public void onSuccess(@Nullable String metadata) {
+        // Use the metadata
+        Log.d("Metadata","The data is" + metadata);
+    }
+
+    @Override
+    public void onError(@NonNull HMSException e) {
+    }
+});
+
+```
+
+</Tab>
+
+## Limitations and Alternatives in the Beta Release
```

Comment:
are you planning to propose any "Alternatives" also (e.g. polling)?

Reply:
just saw the further section - which answers this. can the alternative be a lower-level heading that fits inside this section?

Reply:
Split it kind of